packages/web/src/js/Modal.ts

```diff
@@ -53,13 +58,31 @@ class Modal extends BaseComponent {
     event.preventDefault();
   }
 
+  handleCancel(event: Event) {
+    const { closeOnEscapeKeyDown } = this.config as Config;
+
+    if (closeOnEscapeKeyDown === false) {
+      event.preventDefault();
+    }
+  }
+
+  handleKeyDown(event: KeyboardEvent) {
+    const { closeOnEscapeKeyDown } = this.config as Config;
+
+    if (event.key === EVENT_KEY && closeOnEscapeKeyDown === false && this.isShown) {
+      event.preventDefault();
+    }
+  }
+
   addEventListeners() {
     EventHandler.on(this.element, 'close', (event: KeyboardEvent) => this.onDialogClose(event));
     if (this.isTouchDevice) {
       EventHandler.on(window, 'touchstart', (event: Event & { target: Window }) => this.onClick(event));
     } else {
       EventHandler.on(window, 'click', (event: Event & { target: Window }) => this.onClick(event));
     }
+    EventHandler.on(this.element, 'cancel', (event: Event) => this.handleCancel(event));
+    EventHandler.on(document, 'keydown', (event: KeyboardEvent) => this.handleKeyDown(event));
```

Comment:
shouldn't we mention the Chrome bug here (in this file) as well?

Reply:
comment added